File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        sum=0
        n = len(nums)
        i=0
        j=0
        ans=float('inf')
        while(j<n):
            sum+=nums[j]
            while(sum>=target):
                ans = min(ans,j-i+1)
                sum-=nums[i]
                i+=1
            
            j+=1
        
        if(ans==float('inf')):
            return 0
        return ans
    

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum=0;
        int n = nums.size();
        int i=0,j=0;
        int ans=INT_MAX;
        while(j<n){
            sum+=nums[j];
            while(sum>=target){
                ans = min(ans,j-i+1);
                sum-=nums[i];
                i++;
            }
            
            j++;
        }
        if(ans==INT_MAX)
        return 0;
        return ans;
    }
};